lib/index.ts

```diff
@@ -105,11 +116,92 @@ export class Front {
 			reqType: 'GET' }, params, callback),
 	};
 
+	// Keys for Front access and event verification.
 	private apiKey: string;
+	private apiSecret: string;
 
-	constructor(apiKey: string) {
+	constructor(apiKey: string, apiSecret?: string) {
 		// Key.
 		this.apiKey = apiKey;
+		// Event signature secret.
+		if (apiSecret) {
+			this.apiSecret = apiSecret;
+		}
+	}
+
+	public registerEvents(opts: EventOptions, callback: EventCallback): Server | void {
+		let httpServer: Server| void;
+		let listener: express.Application;
+		const eventQueue: string[] = [];
+		const requestEvent = () => {
+			// Get next event in the queue.
+			const eventId = eventQueue[0];
+			this.httpCall({ path: 'events/<event_id>', reqType: 'GET' }, {
+				event_id: eventId
+			}).then((event: Event) => {
+				callback(null, event);
+			}).catch((err: FrontError) => {
+				callback(err);
+			}).finally(() => {
+				// Get another event if there is one, else finish.
+				eventQueue.shift();
+				if (eventQueue.length > 0) {
+					requestEvent();
+				}
+			});
+		};
+		const addToEventQueue = (id: string): void => {
+			// Push the event onto the queue. If there's nothing already
+			// on it, retrieve now, else just return.
+			eventQueue.push(id);
+			if (eventQueue.length === 1) {
+				requestEvent();
+			}
+		};
+
+		// Ensure we have relevant details to hook into events.
+		if (!this.apiSecret) {
+			throw new Error('No secret key registered');
+		}
+		if (!opts || (opts.server && opts.port) || (!opts.server && !opts.port)) {
+			throw new Error('Pass either an Express instance or a port to listen on');
+		}
+		if (opts.port && typeof opts.port !== 'number') {
+			throw new Error('`port` must be a number');
+		}
+		const hookPath = opts.hookPath || '/fronthook';
+
+		// If we weren't passed an express instance, we create one.
+		if (opts.server) {
+			listener = opts.server;
+		} else {
+			listener = express();
+			httpServer = listener.listen(opts.port);
+		}
+
+		// Use body parser.
+		listener.use(bodyParser.urlencoded({ extended: true }));
+		listener.use(bodyParser.json());
```

Comment:
I've moved the onus onto the client, and mentioned this specifically in the README. If they pass an instance, they have to use middleware that will parse JSON in the body.